llama_index/llms/ollama.py

```diff
@@ -114,10 +114,24 @@ def stream_complete(self, prompt: str, **kwargs: Any) -> CompletionResponseGen:
 
         if not kwargs.get("formatted", False):
             prompt = self.completion_to_prompt(prompt)
+
+        ollama_request_json: Dict[str, Any] = {
+            "prompt": prompt,
+            "model": self.model,
+            "options": all_kwargs,
+        }
+        if all_kwargs.get("system"):
+            ollama_request_json["system"] = all_kwargs["system"]
+            del all_kwargs["system"]
+
+        if all_kwargs.get("formatted"):
+            ollama_request_json["format"] = "json" if all_kwargs["formatted"] else None
+        del all_kwargs["formatted"]
```

Comment:
What happens if `all_kwargs`  does not have 'formatted' key? Won't this result in KeyError?

Reply:
Yes, there are other issues as well. The `query` API is probably broken too.